examples/hashtag-example/src/hashtagStrategy.js

```diff
@@ -0,0 +1,43 @@
+/* @flow */
+// This strategy is from https://github.com/draft-js-plugins/draft-js-plugins/blob/master/draft-js-hashtag-plugin/src/utils/index.js
+import regexes from './hashtagRegex'
+import { ContentBlock } from 'draft-js'
+
+function extractHashtagsWithIndices(text: String): Array {
+  if (!text || !text.match(regexes.hashSigns)) {
+    return []
+  }
+
+  var tags = []
+
+  text.replace(regexes.validHashtag, function(
+    match,
+    before,
+    hash,
+    hashText,
+    offset,
+    chunk
+  ) {
+    const after = chunk.slice(offset + match.length)
+    if (after.match(regexes.endHashtagMatch)) {
+      return
+    }
+    const startPosition = offset + before.length
+    const endPosition = startPosition + hashText.length + 1
+    tags.push({
+      hashtag: hashText,
+      indices: [startPosition, endPosition],
+    })
+  })
+
+  return tags
+}
+
+export default (contentBlock: ContentBlock, callback: Function) => {
+  const text = contentBlock.getText()
+  const results = extractHashtagsWithIndices(text)
```

Comment:
this can be a bit simplified too. The example from the [draft js website should suffice](https://draftjs.org/docs/advanced-topics-decorators):

```
function hashtagStrategy(contentBlock, callback, contentState) {
  findWithRegex(HASHTAG_REGEX, contentBlock, callback);
}

function findWithRegex(regex, contentBlock, callback) {
  const text = contentBlock.getText();
  let matchArr, start;
  while ((matchArr = regex.exec(text)) !== null) {
    start = matchArr.index;
    callback(start, start + matchArr[0].length);
  }
}
```

Btw, this would be a good candidate for the utilities. `findWithRegex` should be in the utilities :) - you're welcome to add it

Reply:
@juliankrispel In order to pass `flow`, i've do some [adjust](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoKBXAdgYwC4CWcWYUBWAJgOoF4AWASgKYDmTAHgBQBOrHAXGGYsAouwAOAGjA4AhjBgAjWTgDWgzgGc8s7nkFYMAW0VNu0ppQPHT3AJRgAvAD4wANzgEKDgN6owMsTaYHgceE5gAOSR-mAwTOFGsng4dACC3NyCAPxaeNzkLADaALp20tq6eADcsQh0BPFgnJxJKemZEbxs7AB0HEw4nKHseHYOAGQTYG2pGdxgAISOjmCGCr6xAZV6EbMd3L3kFBy1AQFyCspqeVUVOrsA1DPJc5lFAAwlvfFYLPR2M5gAC+qGBQA) for `findWithRegex`.